src/sentry/models/identity.py

```diff
@@ -36,11 +36,43 @@ class IdentityProvider(Model):
     """
 
     __include_in_export__ = False
+    organization = FlexibleForeignKey("sentry.Organization", blank=True, null=True)
 
+    # To be renamed to "provider" because this is a bad column name
     type = models.CharField(max_length=64)
+    # Replacing "type"
+    provider = models.CharField(max_length=64, null=True)
+
+    # To be renamed to "provider_id", because we're already renaming "type"
+    external_id = models.CharField(max_length=64, null=True)
+    # Replacing "external_id"
+    provider_id = models.CharField(max_length=64, null=True)
     config = EncryptedJsonField()
+
     date_added = models.DateTimeField(default=timezone.now, null=True)
-    external_id = models.CharField(max_length=64, null=True)
+
+    # SSO-specific fields
+    is_sso = models.BooleanField(default=False)
+    sso_default_team = models.OneToOneField(
+        "sentry.Team", blank=True, null=True, on_delete=models.SET_NULL
+    )
+    sso_default_role = BoundedPositiveIntegerField(default=0)
+    sso_default_global_access = models.BooleanField(default=False)
+    sso_flags = BitField(
+        flags=(
+            ("allow_unlinked", "Grant access to members who have not linked SSO accounts."),
+            ("scim_enabled", "Enable SCIM for member and team provisioning and syncing"),
+        ),
+        default=0,
+    )
```

Comment:
Same comment about defaults here.

Reply:
This is also configuration and should be part of the config object.